src/state/filter.js

```diff
@@ -1,4 +1,31 @@
 import { isExpired, isRemoved } from "./items";
 
-export const filterUncollectedItems = (items) =>
-  items.filter((item) => !isExpired(item) && !isRemoved(item));
+export const filterUncollectedItems = (items, showUncollected) =>
+  items.filter(
+    (item) => showUncollected || (!isExpired(item) && !isRemoved(item))
+  );
+
+export const filterUnmatchedItems = (firstArray, secondArray) =>
```

Comment:
I suspect you could replace this function with a call to lodash's intersection function: https://lodash.com/docs/4.17.15#intersection

Reply:
lodash is so helpful, i need to make it a habit to check lodash before attempting to write any utility function.